model/example_domain_test.go

```diff
@@ -0,0 +1,618 @@
+package model_test
+
+import (
+	"fmt"
+
+	"github.com/nextmv-io/sdk/model"
+)
+
+func ExampleNewDomain_empty() {
+	domain := model.NewDomain()
+
+	fmt.Println(domain.Empty())
+
+	// Output:
+	// true
+}
+
+func ExampleSingleton() {
+	domain := model.Singleton(1)
+
+	fmt.Println(domain.Value())
+
+	// Output:
+	// 1 true
+}
+
+func ExampleMultiple_first() {
+	domain := model.Multiple()
+
+	fmt.Println(domain.Empty())
+
+	// Output:
+	// true
+}
+
+func ExampleMultiple_second() {
+	domain := model.Multiple(3, 5, 1)
+
+	fmt.Println(domain.Min())
+
+	fmt.Println(domain.Max())
+
+	fmt.Println(domain.Len())
+
+	fmt.Println(domain.Contains(3))
+
+	// Output:
+	// 1 true
+	// 5 true
+	// 3
+	// true
+}
```

Comment:
I personally don't like the idea of naming the tests with `first` and `second`, because the rendered godocs use alphabetical ordering so it ends up looking like this (I know this is for a different type but the comment applies here):

<img width="410" alt="image" src="https://user-images.githubusercontent.com/42215971/187501785-30f4f9c5-531e-4d3f-b061-5e2d987409fe.png">

Maybe we can actually unify both tests into a single one, to make it more compact and add a description to show the user what is happening.

```suggestion
// Create an empty domain and one with multiple integer values.
func ExampleMultiple() {
	d1 := model.Multiple()
	fmt.Println(domain.Empty())
	
	d2 := model.Multiple(3, 5, 1)
	fmt.Println(d2.Min())
	fmt.Println(d2.Max())
	fmt.Println(d2.Len())
	fmt.Println(d3.Contains(3))
	// Output:
	// true
	// 1 true
	// 5 true
	// 3
	// true
}
```